crates/subspace-networking/src/request_handlers/generic_request_handler.rs

```diff
@@ -20,33 +20,38 @@ use futures::channel::mpsc;
 use futures::prelude::*;
 use libp2p::PeerId;
 use parity_scale_codec::{Decode, Encode};
+use std::pin::Pin;
 use std::sync::Arc;
 use tracing::{debug, trace};
 
-// Could be changed after the production feedback.
+/// Could be changed after the production feedback.
 const REQUESTS_BUFFER_SIZE: usize = 50;
 
 /// Generic request with associated response
-pub trait GenericRequest: Encode + Decode + 'static {
+pub trait GenericRequest: Encode + Decode + Send + Sync + 'static {
     const PROTOCOL_NAME: &'static str;
     const LOG_TARGET: &'static str;
     /// Response type that corresponds to this request
-    type Response: Encode + Decode + 'static;
+    type Response: Encode + Decode + Send + Sync + 'static;
 }
 
 pub struct GenericRequestHandler<Request: GenericRequest> {
     request_receiver: mpsc::Receiver<IncomingRequest>,
     #[allow(clippy::type_complexity)]
     request_handler: Arc<
-        dyn (Fn(&Request) -> Option<<Request as GenericRequest>::Response>) + Send + Sync + 'static,
+        dyn (Fn(&Request) -> Pin<Box<dyn Future<Output = Option<Request::Response>> + Send>>)
```

Comment:
This requires a type alias:
```
pub type TypeAlias<Request> = Arc<
    dyn (Fn(&Request) -> Pin<Box<dyn Future<Output = Option<<Request as GenericRequest>::Response>> + Send>>)
    + Send
    + Sync
    + 'static,
>;
```
It will allow removing the clippy directive.

Reply:
It will just move it to another line, the type itself needs to be simplified for clippy to shut up, but since it is used just once in this file (I spent some time trying to unify it with `fn create`, but it ended up being less ergonomic than current iteration, so I ended up with this monstrosity).